server/src/weaverbird/pipeline/pipeline.py

```diff
@@ -184,6 +194,79 @@ def dict(self, *, exclude_none: bool = True, **kwargs) -> dict:
 ]
 
 
+def _remove_void_from_combo_condition(
+    condition: ConditionComboAnd | ConditionComboOr | None, combo: str
+) -> ConditionComboAnd | ConditionComboOr | None:
+    """
+    Loop into a combo condition and search for embedded condition that may
+    contains __VOID__ as column/value
+    """
+    if condition is None:
+        return None
+
+    loop_on: list[Condition] = []
+
+    for cond in getattr(condition, combo) or []:
+        if (transformed_condition := _remove_void_from_condition(cond)) is not None:
+            loop_on.append(transformed_condition)
+    if len(loop_on) == 0:
+        return None
+    else:
+        if combo == "and_":
+            condition.and_ = loop_on
+        else:
+            condition.or_ = loop_on
+
+    return condition
+
+
+def _remove_void_from_condition(condition: Condition | None) -> Condition | None:
+    """
+    For a given condition either it's a combo or a simple condition
+    recursivelly check for it's columns and values and remove the one
+    with column/value = __VOID__
+
+    """
+    if isinstance(condition, (ConditionComboAnd, ConditionComboOr)):
+        if isinstance(condition, ConditionComboAnd):
+            condition = _remove_void_from_combo_condition(condition, "and_")
+        else:
+            condition = _remove_void_from_combo_condition(condition, "or_")
+    elif isinstance(
+        condition, (ComparisonCondition, InclusionCondition, MatchCondition, DateBoundCondition)
+    ):
```

Comment:
```suggestion
    elif isinstance(condition, (ComparisonCondition, MatchCondition, DateBoundCondition)):
```
The checks will never be true for an `InclusionCondition`, since it matches against a list of values.

I'd suggest an extra branch:

```python
if isinstance(condition, InclusionCondition):
    # Filter our `__VOID__` from the list and return None in case the resulting list is empty
```

Reply:
handled 👍🏽